internal/run/source.go

```diff
@@ -261,7 +261,7 @@ func (w *Workflow) RunSource(ctx context.Context, parentStep *workflowTracking.W
 
 	if !w.SkipLinting {
 		sourceRes.LintResult, err = w.validateDocument(ctx, rootStep, sourceID, currentDocument, rulesetToUse, w.ProjectDir)
-		if err != nil {
+		if err != nil && !errors.Is(err, validation.ErrValidationFailed) {
```

Comment:
I'm concerned about this line. This change means we will no longer return early if there are linting errors